frontend/src/metabase/selectors/metadata.js

```diff
@@ -202,7 +202,9 @@ export function copyObjects(metadata, objects, instantiate) {
   for (const object of Object.values(objects)) {
     if (object && object.id != null) {
       copies[object.id] = instantiate(object, metadata);
-      copies[object.id].metadata = metadata;
+      if (copies[object.id].metadata == null) {
+        copies[object.id].metadata = metadata;
+      }
```

Comment:
Curious why don't we let it to update metadata here?

Reply:
I've updated this to be a little clearer, but it is because I've added questions to this `Metadata` cache and `Question` has a `metadata` method.